[HIBIKE] More friendly flasher

baby-bell opened the pull request:
A nice help message for the flashing script. Also checks that the device type is valid, and gives a usage message. Fixes issue #391.

baby-bell committed:
Make the flashing script more user friendly

Previously, the flashing script had no help messages and didn't check
that its arguments were valid in any way. This commit adds some
logic to check argument validity and a (somewhat) helpful help
message.

baby-bell committed:
Use devices to find device types

Instead of relying on a fixed list of devices, just scan the
devices directory to figure out which device types exist. This
is more robust to changes.

dHutchings commented on hibike/flash.sh:
```diff
@@ -1,10 +1,57 @@
+#!/bin/bash
 
+NAME="$0"
+SENSOR_TYPES="$(ls devices/)"
```
there's a script to list sensor types, actually: /utils/device_names.sh.

It's true that it does the same thing as line 4.  But it's the script that ubuntu call when it tries to tab-complete.

probably should use the same script.

baby-bell replied:
Thanks for the feedback. Apparently it's bad form to rely on the output of `ls` anyways, so I will change this script to use your script, and your script to use globs.

baby-bell committed:
Delegate finding device names to other script

There is already a script that exists to find device names, so it's
easier to use that. In addition, if the structure of devices changes
at some point, now it is only necessary to change one thing. I have also
changed the device name script to use blobs instead of ls.
